src/test/unit/views/sign-in.test.ts

```diff
@@ -0,0 +1,61 @@
+import { expect } from 'chai';
+import request from 'supertest';
+
+import { app } from '../../../main/app';
+
+const PAGE_URL = '/sign-in';
+const headingClass = 'govuk-fieldset__heading';
+const buttonClass = 'govuk-button';
+const radioClass = 'govuk-radios__item';
+
+const expectedHeader = 'How do you want to sign in?';
+const expectedButtonText = 'Continue';
+const expectedRadioLabel1 = 'Sign in with My HMCTS';
+const expectedRadioLabel2 = 'Sign in with Common Platform';
+const expectedRadioLabel3 = 'Sign in with my P&amp;I details';
+const expectedRadioLabel4 = 'You have signed in before but you\'re not sure which account you used';
+
+let htmlRes: Document;
+describe('Sign In option Page', () => {
+  beforeAll(async () => {
+    await request(app).get(PAGE_URL).then(res => {
+      htmlRes = new DOMParser().parseFromString(res.text, 'text/html');
+    });
+  });
+
+  it('should display header',  () => {
+    const header = htmlRes.getElementsByClassName(headingClass);
+    expect(header[0].innerHTML).contains(expectedHeader, 'Could not find the header');
+  });
+
+  it('should display continue button',  () => {
+    const buttons = htmlRes.getElementsByClassName(buttonClass);
+    expect(buttons[0].innerHTML).contains(expectedButtonText, 'Could not find button');
+  });
+
+  it('should display 4 radio buttons', () => {
+    const radioButtons = htmlRes.getElementsByClassName(radioClass);
+    expect(radioButtons.length).equal(4, '4 radio buttons not found');
+  });
+
+  it('should display first radio button content',  () => {
```

Comment:
this and bellow tests look like something which can be in executed in the for loop, rather than repeating yourself

Reply:
not sure how to do it. Please can you give me an example on how to have the following in a loop?
it('should display first radio button content',  () => {
    const radioButtons = htmlRes.getElementsByClassName(radioClass);
    expect(radioButtons[0].innerHTML).contains(expectedRadioLabel1, 'Could not find the radio button with label ' + expectedRadioLabel1);
  });